core/kazoo_amqp/src/api/kapi_offnet_resource.erl

```diff
@@ -187,7 +187,8 @@ req(JObj) -> req(kz_json:to_proplist(JObj)).
 
 -spec req_v(api_terms()) -> boolean().
 req_v(Prop) when is_list(Prop) ->
-    kz_api:validate(Prop, ?OFFNET_RESOURCE_REQ_HEADERS, ?OFFNET_RESOURCE_REQ_VALUES, ?OFFNET_RESOURCE_REQ_TYPES);
+    kz_api:validate(Prop, ?OFFNET_RESOURCE_REQ_HEADERS, ?OFFNET_RESOURCE_REQ_VALUES, ?OFFNET_RESOURCE_REQ_TYPES)
+        andalso not kz_util:is_empty(props:get_value(?KEY_TO_DID, Prop));
```

Comment:
@jamhed I think @lazedo is saying to add this line over at/around https://github.com/2600hz/kazoo/blob/589a54952a51f35d1f7cf7cd4533c9f3bd13eca7/core/kazoo_amqp/src/api/kapi_offnet_resource.erl#L147